app/vmalert/templates/template.go

```diff
@@ -198,15 +159,16 @@ func FuncsWithQuery(query QueryFn) textTpl.FuncMap {
 	}
 }
 
-// FuncsWithExternalURL returns a function map that depends on externalURL value
-func FuncsWithExternalURL(externalURL *url.URL) textTpl.FuncMap {
+// funcsWithExternalURL returns a function map that depends on externalURL value
+func funcsWithExternalURL(externalURL string) textTpl.FuncMap {
+	eu, _ := url.Parse(externalURL)
```

Comment:
I'd suggest to assert with logger.Panicf() if parsing error is not nil in order to mitigate possible bugs with `nil pointer deref`.

Reply:
Made the function more reliable [here](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/7773/commits/1f4945b86f49f5677e17a5b498fd04f008508761).